### PR TITLE
Fix patches 0013 and 0014 for open issues

### LINKED ISSUE #11: build issue
Hello, can't build the code on 4.13.16
Here is the log:

  CC [M]  /home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali/linux/mali_osk_timers.o
/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali/linux/mali_osk_timers.c: In function ‘_mali_osk_timer_setcallback’:
/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali/linux/mali_osk_timers.c:63:2: error: implicit declaration of function ‘timer_setup’; did you mean ‘ktime_set’? [-Werror=implicit-function-declaration]
  timer_setup(&tim->timer, callback, 0);
  ^~~~~~~~~~~
  ktime_set
cc1: some warnings being treated as errors
scripts/Makefile.build:302: recipe for target '/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali/linux/mali_osk_timers.o' failed
make[2]: *** [/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali/linux/mali_osk_timers.o] Error 1
Makefile:1512: recipe for target '_module_/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali' failed
make[1]: *** [_module_/home/sergey/Projects/drivers/sunxi/orangepipc/sunxi-mali/r6p2/src/devicedrv/mali] Error 2
make[1]: Leaving directory '/home/sergey/Projects/armbian/build/cache/sources/linux-mainline/linux-4.13.y'

### LINKED ISSUE #12: devfreq patch is broken
can't find file to patch at input line 22
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|From 9af33ab01cf84d90fe9a4d354b71dd304bca3010 Mon Sep 17 00:00:00 2001
|From: Maxime Ripard <maxime.ripard@free-electrons.com>
|Date: Thu, 30 Nov 2017 11:45:24 +0100
|Subject: [PATCH] mali: Make devfreq optional
|
|If the USING_DEVFREQ option is set, any device without operating points
|defined in the device tree will be treated as an hard error, making probe
|fail.
|
|Rework the driver a bit to ignore the property missing error, and treat all
|the others as hard errors.
|
|Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
|---
| r6p2/src/devicedrv/mali/linux/mali_devfreq.c | 7 +++++--
| 1 file changed, 5 insertions(+), 2 deletions(-)
|
|diff --git a/r6p2/src/devicedrv/mali/linux/mali_devfreq.c b/r6p2/src/devicedrv/mali/linux/mali_devfreq.c
|index 0b0ba1481eed..c3a2636e6635 100755
|--- a/r6p2/src/devicedrv/mali/linux/mali_devfreq.c
|+++ b/r6p2/src/devicedrv/mali/linux/mali_devfreq.c
--------------------------
File to patch: 
Skip this patch?

### PR DESCRIPTION
**Pull request by giuliobenetti**
@mripard 
Fix 0014 fixes https://github.com/mripard/sunxi-mali/issues/12
Fix 0013 fixes https://github.com/mripard/sunxi-mali/issues/11

I don't know if I've written correctly here.
Is there a standard way when PR multiple commits?

### CODE REVIEW

**Review by sergey-suloev — COMMENTED** [review]
Hey Giulio,
I found that there is "setup_timer" wrapper around "__setup_timer" function. Better use it, I think.

**Comment by giuliobenetti** [comment]
Hi @sergey-suloev,
yes you're right, it would be more elegant and setup_timer wrapper is supported since v4.4.
If Maxime applies my Pull Request, why don't you try to add this little modification?

**Comment by sergey-suloev** [comment]
@giuliobenetti  I forked the branch and can modify what I need. Fast & furious.

**Comment by giuliobenetti** [comment]
Great, now let's wait Maxime to understand how to proceed.

File: patches/0013-mali-support-building-against-4.15.patch [hunk 1]
```diff
@@ -108,7 +108,7 @@ Signed-off-by: John Stultz <john.stultz@linaro.org>
  	MALI_DEBUG_ASSERT_POINTER(tim);
 -	tim->timer.data = (unsigned long)data;
 -	tim->timer.function = (timer_timeout_function_t)callback;
-+	timer_setup(&tim->timer, callback, 0);
++	__setup_timer(&tim->timer, callback, &tim->timer, 0);
```
Comment (mripard) [inline patches/0013-mali-support-building-against-4.15.patch]: Please juste add the definition of timer_setup for the older kernels.

Comment (giuliobenetti) [reply patches/0013-mali-support-building-against-4.15.patch]: Do you mean to place #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0) and use different apis?

Comment (giuliobenetti) [reply patches/0013-mali-support-building-against-4.15.patch]: Or define timer_setup api locally and generate 0015 patch?

Comment (sergey-suloev) [reply patches/0013-mali-support-building-against-4.15.patch]: something like this: 

#if kernel < ver
#define timer_setup(_a, _b, _c) setup_timer(_a, b, blabla, _c)
#endif

**Pull request closed without merging**